Code/Mantid/Framework/Algorithms/src/ProcessIndirectFitParameters.cpp

```diff
@@ -0,0 +1,245 @@
+#include "MantidAlgorithms/ProcessIndirectFitParameters.h"
+
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/TextAxis.h"
+#include "MantidKernel/MandatoryValidator.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+using namespace API;
+using namespace Kernel;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(ProcessIndirectFitParameters)
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+ProcessIndirectFitParameters::ProcessIndirectFitParameters() {}
+
+//----------------------------------------------------------------------------------------------
+/** Destructor
+ */
+ProcessIndirectFitParameters::~ProcessIndirectFitParameters() {}
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string ProcessIndirectFitParameters::name() const {
+  return "ProcessIndirectFitParameters";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int ProcessIndirectFitParameters::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string ProcessIndirectFitParameters::category() const {
+  return "Workflow\\MIDAS";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string ProcessIndirectFitParameters::summary() const {
+  return "Convert a parameter table output by PlotPeakByLogValue to a "
+         "MatrixWorkspace.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void ProcessIndirectFitParameters::init() {
+
+  declareProperty(new WorkspaceProperty<API::ITableWorkspace>(
+                      "InputWorkspace", "", Direction::Input),
+                  "The table workspace to convert to a MatrixWorkspace.");
+
+  declareProperty(
+      "ColumnX", "", boost::make_shared<MandatoryValidator<std::string>>(),
+      "The column in the table to use for the x values.", Direction::Input);
+
+  declareProperty("ParameterNames", "",
+                  boost::make_shared<MandatoryValidator<std::string>>(),
+                  "List of the parameter names to add to the workspace.",
+                  Direction::Input);
+
+  declareProperty("OutputWorkspaceName", "",
```

Comment:
This should be a `MatrixWorkspaceProperty` and the naem shoudl be just `OutputWorkspace`.